web-animations-2/Overview.bs

```diff
@@ -3366,6 +3553,167 @@ Update the type for the AnimationPlaybackEventInit members.
+  attribute TimelineRangeOffset exitRangeEnd;
+};
+</pre>
+
+<div dfn-type=constructor class=constructors dfn-for="AnimationTrigger">
+  : <dfn lt="AnimationTrigger(options)">AnimationTrigger(|options|)</dfn>
+  ::
+      Creates a new {{AnimationTrigger}} object using the following procedure:
+
+      1. Create a new {{AnimationTrigger}} object, |trigger|.
+
+      1. Let |type| be the {{AnimationTriggerOptions/type}} property.
+
+      1. Let |timeline| be the {{AnimationTriggerOptions/timeline}} property.
+
+      1. Run the procedure for [=setting timeline of animation trigger=]
+          on |trigger| passing |timeline| as the <var ignore>new timeline</var>.
+
+      1. Let |range start| be the {{AnimationTriggerOptions/rangeStart}} property.
+
+      1. Let |range end| be the {{AnimationTriggerOptions/rangeEnd}} property.
+
+      1. Set [=default range=] of |trigger| using |range start| and |range end|,
+          following the same rules as the KeyframeAnimationOption
+          <a href="#dom-keyframeanimationoptions-rangestart">rangeStart</a> and
+          <a href="#dom-keyframeanimationoptions-rangeend">rangeEnd</a> respectively.
+
+      1. Let |exit range start| be the {{AnimationTriggerOptions/exitRangeStart}} property.
+
+      1. Let |exit range end| be the {{AnimationTriggerOptions/exitRangeEnd}} property.
+
+      1. Set [=exit range=] of |trigger| using |exit range start| and |exit range end|,
+          following the same rules above for [=default range=], unless both are missing
+          in which case the [=exit range=] is set to the [=default range=].
+
+      <div dfn-type=argument class=parameters
+           dfn-for="AnimationTrigger/AnimationTrigger(options)">
+
+      : <dfn>options</dfn>
+      ::  Configuration parameters for the newly-created trigger.
+
+</div>
+
+<div dfn-type=attribute class=attributes
+       dfn-for="AnimationTrigger">
+
+  : <dfn>timeline</dfn>
+  ::
+      Returns the [=timeline=] for this trigger
+      or `null` if this timeline is [=inactive timeline|inactive=].
+
+  : <dfn>type</dfn>
+  ::
+      Returns the [=animation trigger type|type=] of this trigger.
+
+  : <dfn>rangeStart</dfn>
+  ::
+      Returns the [=default range=]’s start of this trigger.
+
+  : <dfn>rangeEnd</dfn>
+  ::
+      Returns the [=default range=]’s end of this trigger.
+
+  : <dfn>exitRangeStart</dfn>
+  ::
+      Returns the [=exit range=]’s start of this trigger.
+
+  : <dfn>exitRangeEnd</dfn>
+  ::
+      Returns the [=exit range=]’s end of this trigger.
+
+</div>
+
+<h3 id="the-animationtriggeroptions-dictionary">The <code>AnimationTriggerOptions</code> dictionary</h3>
+
+<pre class='idl'>
+dictionary AnimationTriggerOptions {
+  AnimationTimeline? timeline;
+  AnimationTriggerType? type = "once";
+  TimelineRangeOffset? rangeStart = "normal";
+  TimelineRangeOffset? rangeEnd = "normal";
+  TimelineRangeOffset? exitRangeStart;
+  TimelineRangeOffset? exitRangeEnd;
```

Comment:
I think rather than being nullable these all have default values just as their css counterparts do. Note that [TimelineRangeOffset](https://drafts.csswg.org/web-animations-2/#dictdef-timelinerangeoffset) is a structure and can't hold string values like "normal", but ([TimelineRangeOffset](https://drafts.csswg.org/web-animations-2/#dictdef-timelinerangeoffset) or [CSSNumericValue](https://drafts.css-houdini.org/css-typed-om-1/#cssnumericvalue) or [CSSKeywordValue](https://drafts.css-houdini.org/css-typed-om-1/#csskeywordvalue) or [DOMString](https://webidl.spec.whatwg.org/#idl-DOMString)) can.

Reply:
The thing about the exit range, is that you want it to by default inherit the default range values, and not reset to `normal`.
Maybe we should specify that with an `auto` value that will be the default?

Reply:
Oh, we already have that in the css counterparts, so adding it here as well.

Reply:
done